src/nvofapi/nvofapi_vulkan.cpp

```diff
@@ -0,0 +1,168 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: MIT
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+#include "util/util_log.h"
+#include "util/util_string.h"
+
+#include "../inc/nvofapi/nvOpticalFlowVulkan.h"
+
+#include "nvofapi.h"
+
+namespace nvofapi {
+
+    bool NvOFInstanceVk::Initialize() {
+        m_library = LoadLibraryA("winevulkan.dll");
```

Comment:
While for D3D12 it does make sense to load VK funcs from winevulkan directly (because vkd3d-proton does the same thing), doesn't doing the same for native Vulkan OFA risk a mismatch if there's native Vulkan loader installed in the Wine/Proton prefix? Will funcs exported by `winevulkan.dll` be able to handle VK handles that were created using functions exported by `vulkan-1.dll` if that's the library the application loaded an is using to access Vulkan API?